linux/ELF.c

```diff
@@ -0,0 +1,512 @@
+#include "ELF.h"
+
+#ifdef HAVE_LIBELF
+
+#include <errno.h>
+#include <gelf.h>
+
+#include "CRT.h"
+#include "Hashtable.h"
+
+
+static Hashtable* resultCache = NULL;
+
+void ELF_init() {
+   if (!resultCache)
+      resultCache = Hashtable_new(100, true);
+}
+
+void ELF_cleanup() {
+   if (resultCache) {
+      Hashtable_delete(resultCache);
+      resultCache = NULL;
+   }
+}
+
+struct CacheEntry {
+   ino_t st_ino;
+   dev_t st_dev;
+   off_t st_size;
+   elf_state_t data;
+   char runpath[256];
```

Comment:
Limited by origin data structure or candidate for proper `strdup`/`free` handling via `char*`?